ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.hpp

```diff
@@ -148,7 +148,8 @@ AllGather create_all_gather_struct(
     const std::optional<MemoryConfig>& memory_config,
     const std::optional<size_t> user_defined_num_workersm,
     const std::optional<size_t> user_defined_num_buffers_per_channel,
-    const std::vector<Device*>& devices
+    const std::vector<Device*>& devices,
+    const all_gather_op::Topology topology = all_gather_op::Topology::Ring
```

Comment:
I think this is the only thing we might want to change. I think we can force the caller to specify the topology now at this point with your changes.